multiapps-controller-process/src/main/java/org/cloudfoundry/multiapps/controller/process/util/StagingApplicationAttributeUpdater.java

```diff
@@ -3,46 +3,90 @@
+import com.sap.cloudfoundry.client.facade.domain.DockerData;
+import com.sap.cloudfoundry.client.facade.domain.DockerInfo;
+import com.sap.cloudfoundry.client.facade.domain.LifecycleType;
 import com.sap.cloudfoundry.client.facade.domain.Staging;
+import com.sap.cloudfoundry.client.facade.domain.ImmutableDockerCredentials;
+import com.sap.cloudfoundry.client.facade.domain.ImmutableDockerInfo;
 
 public class StagingApplicationAttributeUpdater extends ApplicationAttributeUpdater {
 
-    public StagingApplicationAttributeUpdater(Context context) {
+    private final CloudProcess existingProcess;
+
+    public StagingApplicationAttributeUpdater(Context context, CloudProcess process) {
         super(context, UpdateStrategy.REPLACE);
+        this.existingProcess = process;
     }
 
     @Override
-    protected boolean shouldUpdateAttribute(CloudApplication existingApplication, CloudApplication application) {
+    @SuppressWarnings("unchecked")
+    protected boolean shouldUpdateAttribute(CloudApplication existingApplication, CloudApplicationExtended application) {
         Staging staging = application.getStaging();
-        Staging existingStaging = existingApplication.getStaging();
-        return hasStagingChanged(staging, existingStaging);
+        boolean isSshEnabled = getControllerClient().getApplicationSshEnabled(existingApplication.getGuid());
+        DockerInfo dockerInfo = null;
+        List<String> buildpacks = null;
+        String stack = null;
+        var lifecycle = existingApplication.getLifecycle();
+        if (lifecycle.getType() == LifecycleType.BUILDPACK) {
+            buildpacks = (List<String>) lifecycle.getData()
+                                                 .get("buildpacks");
+            stack = (String) lifecycle.getData()
+                                      .get("stack");
+        } else {
+            var droplet = getControllerClient().getCurrentDropletForApplication(existingApplication.getGuid());
+            var cloudPackage = getControllerClient().getPackage(droplet.getPackageGuid());
+            var dockerData = (DockerData) cloudPackage.getData();
+            dockerInfo = ImmutableDockerInfo.builder()
+                                            .image(dockerData.getImage())
+                                            .build();
+            if (dockerData.getUsername() != null && dockerData.getPassword() != null) {
+                dockerInfo = ImmutableDockerInfo.copyOf(dockerInfo)
+                                                .withCredentials(ImmutableDockerCredentials.builder()
+                                                                                           .username(dockerData.getUsername())
+                                                                                           .password(dockerData.getPassword())
+                                                                                           .build());
+            }
+        }
```

Comment:
Could you create some comparators, one for buildpack and one for docker, and remove this if - else

Reply:
+1